### PR TITLE
feat(vpn): enhance customer gateway resource with id_type, id_value, ca_certificate.id attributes and deprecate route_mode, ip attributes

### PR DESCRIPTION
**Pull request by profoundwu**
**What this PR does / why we need it**:
enhance customer gateway resource with id_type, id_value, ca_certificate.id attributes and deprecate route_mode, ip attributes
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. enhance customer gateway resource with id_type, id_value, ca_certificate.id attributes and deprecate route_mode, ip attributes
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/vpn" TESTARGS="-run TestAccCustomerGateway"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/vpn -v -run TestAccCustomerGateway -timeout 360m -parallel 4
=== RUN   TestAccCustomerGateway_basic_withDeprecatedFields
=== PAUSE TestAccCustomerGateway_basic_withDeprecatedFields
=== RUN   TestAccCustomerGateway_certificate_withDeprecatedFields
=== PAUSE TestAccCustomerGateway_certificate_withDeprecatedFields
=== RUN   TestAccCustomerGateway_basic
=== PAUSE TestAccCustomerGateway_basic
=== RUN   TestAccCustomerGateway_certificate
=== PAUSE TestAccCustomerGateway_certificate
=== CONT  TestAccCustomerGateway_basic_withDeprecatedFields
=== CONT  TestAccCustomerGateway_basic
=== CONT  TestAccCustomerGateway_certificate
=== CONT  TestAccCustomerGateway_certificate_withDeprecatedFields
--- PASS: TestAccCustomerGateway_basic_withDeprecatedFields (48.21s)
--- PASS: TestAccCustomerGateway_basic (48.67s)
--- PASS: TestAccCustomerGateway_certificate_withDeprecatedFields (48.84s)
--- PASS: TestAccCustomerGateway_certificate (48.96s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/vpn       49.013s

```

### CODE REVIEW

File: docs/resources/vpn_customer_gateway.md [hunk 1]
```diff
@@ -46,17 +46,19 @@ The following arguments are supported:
 
 * `name` - (Required, String) The customer gateway name.
 
-* `ip` - (Required, String, ForceNew) The IP address of the customer gateway.
+* `id_value` - (Optional, String, ForceNew) Specifies the identifier of a customer gateway.
+  When id_type is set to **ip**, the value is an IPv4 address in dotted decimal notation, for example, 192.168.45.7.
```
Comment (saf3dfsa) [inline docs/resources/vpn_customer_gateway.md]:
```suggestion
  When `id_type` is set to **ip**, the value is an IPv4 address in dotted decimal notation, for example, 192.168.45.7.
```

File: huaweicloud/services/acceptance/vpn/resource_huaweicloud_vpn_customer_gateway_test.go [hunk 2]
```diff
@@ -167,23 +285,59 @@ resource "huaweicloud_vpn_customer_gateway" "test" {
...
+  id_value = "%s"
+
+  tags = {
+    key = "val"
+    foo = "bar-update"
+  }
+}`, name, ipAddress)
+}
+
+func testCustomerGateway_certificate(name, ipAddress string, certificateContent string) string {
```
Comment (saf3dfsa) [inline huaweicloud/services/acceptance/vpn/resource_huaweicloud_vpn_customer_gateway_test.go]:
```suggestion
func testCustomerGateway_certificate(name, ipAddress, certificateContent string) string {
```

File: huaweicloud/services/vpn/resource_huaweicloud_vpn_customer_gateway.go [hunk 3]
```diff
@@ -166,13 +196,20 @@ func resourceCustomerGatewayCreate(ctx context.Context, d *schema.ResourceData,
 }
 
 func buildCreateCustomerGatewayBodyParams(d *schema.ResourceData) map[string]interface{} {
```
Comment (saf3dfsa) [inline huaweicloud/services/vpn/resource_huaweicloud_vpn_customer_gateway.go:203]: You can change it into one function, buildCreateCustomerGatewayDeprecatedCustomerGatewayChildBody
and buildCreateCustomerGatewayCustomerGatewayChildBody is unnecessary.

File: huaweicloud/services/acceptance/vpn/resource_huaweicloud_vpn_customer_gateway_test.go [hunk 4]
```diff
@@ -87,6 +201,9 @@ func TestAccCustomerGateway_basic(t *testing.T) {
 				ResourceName:      rName,
 				ImportState:       true,
 				ImportStateVerify: true,
+				ImportStateVerifyIgnore: []string{
+					"route_mode",
```
Comment (saf3dfsa) [inline huaweicloud/services/acceptance/vpn/resource_huaweicloud_vpn_customer_gateway_test.go:205]: you didn't input `route_mode`, and didn't set when read, why you have to ignore it?

File: docs/resources/vpn_customer_gateway.md [hunk 5]
```diff
@@ -46,17 +46,19 @@ The following arguments are supported:
 
 * `name` - (Required, String) The customer gateway name.
 
-* `ip` - (Required, String, ForceNew) The IP address of the customer gateway.
+* `id_value` - (Optional, String, ForceNew) Specifies the identifier of a customer gateway.
```
Comment (Jason-Zhang9309) [inline docs/resources/vpn_customer_gateway.md]:
```suggestion
* `id_value` - (Required, String, ForceNew) Specifies the identifier of a customer gateway.
```

File: huaweicloud/services/vpn/resource_huaweicloud_vpn_customer_gateway.go [hunk 6]
```diff
@@ -56,34 +56,64 @@ func ResourceCustomerGateway() *schema.Resource {
 					validation.StringLenBetween(1, 64),
 				),
 			},
+			"id_value": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ForceNew:      true,
+				Computed:      true,
+				Description:   `The identifier of a customer gateway.`,
```
Comment (Jason-Zhang9309) [inline huaweicloud/services/vpn/resource_huaweicloud_vpn_customer_gateway.go]:
```suggestion
				Description: utils.SchemaDesc(
								"The identifier of a customer gateway.",
								utils.SchemaDescInput{
									Required: true,
								},
							),
```

**Pull request merged**